src/us/kbase/workspace/test/kbase/SchemaUpdaterCLITest.java

```diff
@@ -0,0 +1,271 @@
+		
+		final int exitCode = SchemaUpdaterCLI.run(mocks.updater, mocks.provider, out, err, args);
+		
+		assertThat("incorrect sys err", err.toString(), is(expectedErr));
+		assertThat("incorrect sys out", out.toString(), is(expectedOut));
+		assertThat("incorrect exit code", exitCode, is(expectedExitCode));
+	}
+	
+	public String generateHelpString(final List<String> preface) {
+		final LinkedList<String> helpCopy = new LinkedList<String>(HELP);
+		Collections.reverse(preface);
+		preface.forEach(i -> helpCopy.addFirst(i));
+		return String.join("\n", helpCopy) + "\n";
+	}
+	
+	@Test
+	public void runNoArgs() throws Exception {
+		run(array(VALID_CONFIG_FILE.toString()), false, false);
+	}
+	
+	@Test
+	public void runSingleCharArgs() throws Exception {
+		run(array("-c", VALID_CONFIG_FILE.toString(), "-o"), true, true);
+	}
+	
+	@Test
+	public void runFullLengthArgs() throws Exception {
+		run(array("--override-version-check", VALID_CONFIG_FILE.toString(), "--complete"),
+				true, true);
+	}
+	
+	private void run(final String[] args, final boolean complete, final boolean override)
+			throws Exception {
+		final Mocks m = new Mocks();
+		
+		when(m.provider.provide(new KBaseWorkspaceConfig(VALID_CONFIG))).thenReturn(m.client);
+		when(m.updater.update(eq(m.db), any(), eq(complete), eq(override))).then(new UpdateAnswer(
+				list("I'm, like, supposed to find some types or something?", "whatever")));
+		
+		runUpdater(m, args, "I'm, like, supposed to find some types or something?\nwhatever\n");
+	}
+	
+	@Test
+	public void version() throws Exception {
+		version(array("-V"));
+		version(array("--version"));
+	}
+	
+	private void version(final String[] args) {
+		runUpdater(new Mocks(), args, VERSION + "\n");
+	}
+
+	@Test
+	public void help() throws Exception {
+		help(array("-h"));
+		help(array("--help"));
+	}
+	
+	private void help(final String[] args) {
+		runUpdater(new Mocks(), args, String.join("\n", HELP) + "\n");
+	}
+
+	@Test
+	public void badOptionFail() throws Exception {
+		final String err =
+				"Unknown option: '--o'\nPossible solutions: -o, --override-version-check\n";
+		runUpdater(new Mocks(), array("--o", VALID_CONFIG_FILE.toString()), "", err, 2);
+	}
+	
+	@Test
+	public void noArgsFail() throws Exception {
+		final String err = generateHelpString(list("Missing required parameter: '<CONFIG_FILE>'"));
+		runUpdater(new Mocks(), array(), "", err, 2);
+	}
+	
+	@Test
+	public void tooManyPositionalsFail() throws Exception {
+		final String err = generateHelpString(list("Unmatched argument at index 1: 'file2'"));
+		runUpdater(new Mocks(), array("file1", "file2"), "", err, 2);
+	}
+	
+	@Test
+	public void noConfigFileFail() throws Exception {
+		final Path file = Paths.get("super_fake_file");
+		final String err = generateHelpString(list(String.format(
+				"Could not read from configuration file super_fake_file: %s " +
+						"(No such file or directory)",
+				file.toAbsolutePath())));
```

Comment:
What would the error be if the file exists but cannot be read (e.g. if it has the wrong permissions)?

Reply:
```
Could not read from configuration file foo: /home/crushingismybusiness/github/mrcreosote/workspace_deluxe/foo (Permission denied)
Usage: update_workspace_database_schema [-choV] <CONFIG_FILE>
...
```